### ISSUE TITLE
Check failed: cusolverDnCreate(&cusolver_dn_handle) == CUSOLVER_STATUS_SUCCESS Failed to create cuSolverDN instance.

### ISSUE THREAD

**Issue by geovsion** [rpautrat/SuperPoint#225]
(superpoint) lhj@zhouwei-SYS-4028GR-TR2:~/SuperPoint/superpoint$ python export_detections.py configs/magic-point_coco_export.yaml magic-point_synth --pred_only --batch_size=5 --export_name=magic-point_coco-export1
export_detections.py:25: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  config = yaml.load(f)
[07/30/2021 11:51:55 INFO] Number of GPUs detected: 2
[07/30/2021 11:51:57 WARNING] From /home/lhj/SuperPoint/superpoint/datasets/utils/pipeline.py:96: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
[07/30/2021 11:51:57 WARNING] From /home/lhj/SuperPoint/superpoint/datasets/utils/pipeline.py:98: to_int32 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
[07/30/2021 11:51:57 WARNING] From /home/lhj/.conda/envs/superpoint/lib/python3.6/site-packages/tensorflow/python/ops/control_flow_ops.py:3632: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
2021-07-30 11:52:00.697226: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
2021-07-30 11:52:00.728028: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2500025000 Hz
2021-07-30 11:52:00.731226: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x5fcebc0 executing computations on platform Host. Devices:
2021-07-30 11:52:00.731269: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): <undefined>, <undefined>
2021-07-30 11:52:06.486409: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1433] Found device 0 with properties: 
name: GeForce RTX 2080 Ti major: 7 minor: 5 memoryClockRate(GHz): 1.545
pciBusID: 0000:04:00.0
totalMemory: 10.76GiB freeMemory: 10.61GiB
2021-07-30 11:52:06.571882: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1433] Found device 1 with properties: 
name: GeForce RTX 2080 Ti major: 7 minor: 5 memoryClockRate(GHz): 1.545
pciBusID: 0000:06:00.0
totalMemory: 10.76GiB freeMemory: 10.61GiB
2021-07-30 11:52:06.571961: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1512] Adding visible gpu devices: 0, 1
2021-07-30 11:52:06.573498: I tensorflow/core/common_runtime/gpu/gpu_device.cc:984] Device interconnect StreamExecutor with strength 1 edge matrix:
2021-07-30 11:52:06.573512: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990]      0 1 
2021-07-30 11:52:06.573518: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 0:   N N 
2021-07-30 11:52:06.573522: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 1:   N N 
2021-07-30 11:52:06.573634: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 10320 MB memory) -> physical GPU (device: 0, name: GeForce RTX 2080 Ti, pci bus id: 0000:04:00.0, compute capability: 7.5)
2021-07-30 11:52:06.573891: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:1 with 10320 MB memory) -> physical GPU (device: 1, name: GeForce RTX 2080 Ti, pci bus id: 0000:06:00.0, compute capability: 7.5)
2021-07-30 11:52:06.575495: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0xb518750 executing computations on platform CUDA. Devices:
2021-07-30 11:52:06.575514: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): GeForce RTX 2080 Ti, Compute Capability 7.5
2021-07-30 11:52:06.575531: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (1): GeForce RTX 2080 Ti, Compute Capability 7.5
[07/30/2021 11:52:06 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:06 WARNING] From /home/lhj/SuperPoint/superpoint/models/backbones/vgg.py:10: conv2d (from tensorflow.python.layers.convolutional) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.conv2d instead.
[07/30/2021 11:52:06 WARNING] From /home/lhj/SuperPoint/superpoint/models/backbones/vgg.py:14: batch_normalization (from tensorflow.python.layers.normalization) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.batch_normalization instead.
[07/30/2021 11:52:06 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:06 WARNING] From /home/lhj/SuperPoint/superpoint/models/backbones/vgg.py:28: max_pooling2d (from tensorflow.python.layers.pooling) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.max_pooling2d instead.
[07/30/2021 11:52:06 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:06 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:06 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:06 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:06 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:06 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:06 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:06 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:07 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:07 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:07 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:07 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:07 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:07 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:07 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:07 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:07 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:07 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:07 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:07 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:07 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:07 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:07 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:07 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:07 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:07 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:07 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:07 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:08 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:08 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:08 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:08 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:08 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:08 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:08 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:08 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:08 INFO] Scale of 0 disables regularizer.
[07/30/2021 11:52:08 INFO] Scale of 0 disables regularizer.
2021-07-30 11:52:08.777575: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1512] Adding visible gpu devices: 0, 1
2021-07-30 11:52:08.777638: I tensorflow/core/common_runtime/gpu/gpu_device.cc:984] Device interconnect StreamExecutor with strength 1 edge matrix:
2021-07-30 11:52:08.777649: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990]      0 1 
2021-07-30 11:52:08.777656: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 0:   N N 
2021-07-30 11:52:08.777661: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 1:   N N 
2021-07-30 11:52:08.777784: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 10320 MB memory) -> physical GPU (device: 0, name: GeForce RTX 2080 Ti, pci bus id: 0000:04:00.0, compute capability: 7.5)
2021-07-30 11:52:08.777958: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:1 with 10320 MB memory) -> physical GPU (device: 1, name: GeForce RTX 2080 Ti, pci bus id: 0000:06:00.0, compute capability: 7.5)
[07/30/2021 11:52:10 WARNING] From /home/lhj/.conda/envs/superpoint/lib/python3.6/site-packages/tensorflow/python/training/saver.py:1266: checkpoint_exists (from tensorflow.python.training.checkpoint_management) is deprecated and will be removed in a future version.
Instructions for updating:
Use standard file APIs to check for files with this prefix.
[07/30/2021 11:52:10 INFO] Restoring parameters from /home/lhj/SuperPoint/experiments/magic-point_synth/model.ckpt-50000
0it [00:00, ?it/s]
0it [00:00, ?it/s]2021-07-30 11:52:17.198647: I tensorflow/stream_executor/dso_loader.cc:152] successfully opened CUDA library libcublas.so.10.0 locally
2021-07-30 11:52:19.556017: I tensorflow/core/kernels/cuda_solvers.cc:159] Creating CudaSolver handles for stream 0x15952e30
2021-07-30 11:52:20.638335: F tensorflow/core/kernels/cuda_solvers.cc:94] Check failed: cusolverDnCreate(&cusolver_dn_handle) == CUSOLVER_STATUS_SUCCESS Failed to create cuSolverDN instance.
Aborted (core dumped)

tensorflow=1.13.1 and python=3.6.1, cuda=10.1.  I don't know how to solve above problem, can you give me some advice? thanks!

**Comment by rpautrat** [comment]
Hi, this is probably due to a wrong compatibility between the code, TF and CUDA versions. Have you tried the configuration suggested in this issue: https://github.com/rpautrat/SuperPoint/issues/173#issue-730896838? It worked for many people at least.

**Comment by rpautrat** [comment]
I am closing this now, feel free to reopen it if necessary.
